sdk/resourcemanager/Azure.ResourceManager/src/ArmOperation.cs

```diff
@@ -3,13 +3,78 @@
 
 // <auto-generated/>
 
-#nullable disable
+#nullable enable
 
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Azure.Core;
+using Azure.Core.Pipeline;
 
 namespace Azure.ResourceManager
 {
     /// <inheritdoc/>
-    public abstract class ArmOperation : Operation
-    {
+    public class ArmOperation : Operation
+{
+        private readonly OperationInternal? _operation;
+        private readonly RehydrationToken? _rehydrationToken;
+
+        /// <summary> Initializes a new instance of ArmOperation for mocking. </summary>
+        protected ArmOperation()
+        {
+        }
+
+        internal ArmOperation(ClientDiagnostics clientDiagnostics, HttpPipeline pipeline, Request request, Response response, OperationFinalStateVia finalStateVia, bool skipApiVersionOverride = false, string? apiVersionOverrideValue = null)
+        {
+            var nextLinkOperation = NextLinkOperationImplementation.Create(pipeline, request.Method, request.Uri.ToUri(), response, finalStateVia, skipApiVersionOverride, apiVersionOverrideValue);
+            _operation = new OperationInternal(nextLinkOperation, clientDiagnostics, response);
+        }
+
+        /// <summary> Create an instance of the <see cref="ArmOperation"/> class from rehydration. </summary>
+        /// <param name="client"> The client parameters to use in these operations. </param>
+        /// <param name="rehydrationToken"> The id of the ArmOperation. </param>
+        public ArmOperation(ArmClient client, RehydrationToken? rehydrationToken)
+        {
+            Argument.AssertNotNull(rehydrationToken, nameof(rehydrationToken));
+            Argument.AssertNotNull(client, nameof(client));
+
+            var nextLinkOperation = NextLinkOperationImplementation.Create(client.Pipeline, rehydrationToken);
+            // TODO: Do we need more specific OptionsNamespace, ProviderNamespace and OperationTypeName and possibly from id?
+            var clientDiagnostics = new ClientDiagnostics("Azure.ResourceManager", "Microsoft.Resources", client.Diagnostics);
```

Comment:
Are these the correct diagnostics for every LRO we are rehydrating?

Reply:
Will update to `new ClientDiagnostics(ClientOptions.Default)`

Reply:
to check if DPG has this default

Reply:
[This](https://github.com/Azure/azure-sdk-for-net/blob/c42894e46a2c0c98b678494f59a0031fbe18986d/sdk/core/Azure.Core/src/ClientOptions.cs#L23) is available for DPG as well.